app/models/concerns/pending_changes_concern.rb

```diff
@@ -6,17 +6,33 @@ module PendingChangesConcern
   included { attribute :pending_changes, :jsonb, default: {} }
 
   def stage_changes(attributes)
```

Comment:
I think there's potential to refactor this method as I'm finding it a bit complicated/hard to read. I've got a suggestion below which I think works and I find easier to understand:

```rb
def stage_changes(attributes)
  attributes.each do |attr, new_value|
    current_value = public_send(attr)

    if normalised(new_value) == normalised(current_value)
      public_send("#{attr}=", new_value)
      pending_changes.delete(attr.to_s)
    else
      pending_changes[attr.to_s] = new_value
    end
  end

  save!
end

private

def normalised(value)
  value.respond_to?(:downcase) ? value.downcase : value
end
```

What do you think?